Tests/LogtoClientTests/LogtoAuthSession/LogtoWebViewAuthSessionTests.swift

```diff
@@ -0,0 +1,100 @@
+//
+//  LogtoWebViewAuthSessionTests.swift
+//
+//
+//  Created by Gao Sun on 2022/4/14.
+//
+
+import Foundation
+@testable import LogtoClient
+import WebKit
+import XCTest
+
+class LogtoWebViewAuthSessionMock: LogtoWebViewAuthSession {
+    let viewControllerMock = UIViewController()
+
+    override func getTopViewController() -> UnifiedViewController? {
+        viewControllerMock
+    }
+}
+
+final class WKNavigationActionMock: WKNavigationAction {
+    override var request: URLRequest { urlRequest }
+    let urlRequest: URLRequest
+
+    init(urlRequest: URLRequest) {
+        self.urlRequest = urlRequest
+        super.init()
+    }
+}
+
+final class LogtoWebViewAuthSessionTests: XCTestCase {
+    let mockUrl = URL(string: "https://logto.dev")!
+
+    func testStartFailed() {
+        let session = LogtoWebViewAuthSession(mockUrl, redirectUri: mockUrl, socialPlugins: []) { _ in }
```

Comment:
A little bit not following on these two test cases.

With the original unmocked `LogtoWebViewAuthSession` class,  will `getTopViewController` returns a undefined controller? Is it a valid case in production? 
Should we rename the teat to sth like, session should not start if no viewController found?

Reply:
yes, let me update